### DISCUSSION TITLE
New Transports

### DISCUSSION THREAD

**Discussion by visionm45** [boardgameio/boardgame.io#1174] [Ideas]
HI.  I have been considering a few different things using boardgame.io. currently I am using the framework in an unconventional way and because of this I am not able to use boardgame.io to its full potential. 

 It seems to me that in the "correct" way for me to do what I am thinking would be to write a custom transport.  I have gone back and forth on the thought of actually doing this and so I would like to explore this possibility a bit more.  Is there a standard way of doing things or any docs regarding transports? or would I just basically need to look at the current implementation of the local and socketio and follow the example of how things were done there?

as an aside.  I made a comment on a feature request for a [p2p transport](https://github.com/boardgameio/boardgame.io/issues/580) because I couldn't find anything related anywhere else.

**Comment by customautosys** [comment]
Isn't this what you want? https://github.com/boardgameio/p2p
